subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperGenerationIntegrationTest.groovy

```diff
@@ -59,6 +60,22 @@ class WrapperGenerationIntegrationTest extends AbstractIntegrationSpec {
         file("gradle/wrapper/gradle-wrapper.properties").text.contains("distributionUrl=https\\://services.gradle.org/distributions/gradle-2.2.1-bin.zip")
     }
 
+    def "generated wrapper scripts for given distribution type from command-line"() {
+        when:
+        run "wrapper", "--gradle-version", "2.13", "--dist-type", "all"
+
+        then:
+        file("gradle/wrapper/gradle-wrapper.properties").text.contains("distributionUrl=https\\://services.gradle.org/distributions/gradle-2.13-all.zip")
+    }
+
+    def "generated wrapper scripts for given distribution type from command-line"() {
+        when:
+        run "wrapper", "--gradle-version", "2.13", "--dist-type", "invalid-distribution-type"
+
+        then:
+        thrown(CommandLineArgumentException)
```

Comment:
@bmuschko I've updated the PR, but with the current code I get

    java.lang.AssertionError: 
    Expected: a collection containing an exception with message that is a string starting with "Invalid distribution type \"invalid-distribution-type\"." (normalize line separators)
         but: 
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.gradle.integtests.fixtures.executer.InProcessGradleExecuter$InProcessExecutionFailure.assertThatCause(InProcessGradleExecuter.java:516)
        at org.gradle.integtests.fixtures.executer.InProcessGradleExecuter$InProcessExecutionFailure.assertHasCause(InProcessGradleExecuter.java:509)
        at org.gradle.integtests.WrapperGenerationIntegrationTest.generated wrapper scripts for given distribution type from command-line(WrapperGenerationIntegrationTest.groovy:76)

What am I doing wrong?